commit 2dcbfa0c9de95499902a2313d70ec4422b020770
Author: Andreas Pakulat <apaku@gmx.de>
Date:   Wed May 19 09:36:47 2010 +0200

    Use the complete url of the active document.
    
    Don't just use the directory, use the complete url of the active
    document and go up one step. This fixes the default-dir when working
    with remote url.
    
    BUG:238094

diff --git a/shell/documentcontroller.cpp b/shell/documentcontroller.cpp
index 2288972..6d9344f 100644
--- a/shell/documentcontroller.cpp
+++ b/shell/documentcontroller.cpp
@@ -176,7 +176,7 @@ struct DocumentControllerPrivate {
             KUrl dir;
             if( controller->activeDocument() )
             {
-                dir = KUrl( controller->activeDocument()->url().directory() );
+                dir = controller->activeDocument()->url().upUrl();
             }else
             {
                 dir = KGlobal::config()->group("Open File").readEntry( "Last Open File Directory", Core::self()->projectController()->projectsBaseDirectory() );
